pyproject.toml

```diff
@@ -1,5 +1,40 @@
+[build-system]
+requires = ["setuptools", "setuptools-scm"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "mkosi"
+authors = [
+    {name = "mkosi contributors", email = "systemd-devel@lists.freedesktop.org"},
+]
+version = "14"
+description = "Build Bespoke IS Images"
+readme = "README.md"
+requires-python = ">=3.9"
+license = {file = "LICENSE"}
+
+[project.optional-dependencies]
+cryptography = ["cryptography"]
+dev = [
+    "pytest",
+    "mypy",
+    "types-cryptography",
+    "isort",
+    "pyflakes",
+]
+
+[project.scripts]
+mkosi = "mkosi.__main__:main"
+
+[tool.setuptools]
+packages = ["mkosi"]
```

Comment:
Haven't checked this yet, but this seems to be mostly needed because we don't keep the `mkosi` module in ` src/` directory, which seems to be what all the cool kids do nowadays.

Reply:
Can we rename it to src?

Reply:
I think that's more of a [move](https://packaging.python.org/en/latest/discussions/src-layout-vs-flat-layout/), i.e. `mkosi/ -> src/mkosi/`. I don't think that would be a problem and I don't have strong preferences here.

Reply:
What is the point of the move? This makes development a bit more inconvenient, so if there's a reason to do it, it should be provided.

Reply:
I'm lukewarm on changing this. It looks like it's more and more the default for Python projects nowadays and tools seem to target this as the default. The reason seems to be because people actually want to avoid being able to import something by accident. I don't really see this as a problem and am fine with keeping this as is, I only wanted to note here why this looks the way it does and what could be done about it.